php/sms-template-detector/v2/app/SmsTemplateDetector.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace App;
+
+class SmsTemplateDetector
+{
+    /**
+     * @var array The templates to detect sms pattern.
+     */
+    protected $templates = [
+        'Purchase of {amount} with {card} at {brand},',
+        'Payment of {amount} to {brand} with {card}.',
+    ];
+
+    /**
+     * @var array The template replacements to be extracted form sms.
+     */
+    protected $templateReplacemnts = [
+        '{amount}' => '(.*?)',
+        '{brand}' => '(.*?)',
+        '{card}' => '(.*?)',
+        '{account}' => '(.*?)',
+        '{datetime}' => '(.*?(?=\.))',
+    ];
+
+    /**
+     * Retrieves the templates and it can be overidden with custom logic.
+     * 
+     * @return array
+     */
+    public function getTemplates()
```

Comment:
do we need to have this method public?

Reply:
Hi @saleem-hadad 

Declaring the getTemplates() method as public allows external code to directly access and retrieve the templates which breaks encapsulation principle, but I intended the method getTemplates() to be used as part of library. It allows users of the class to easily retrieve the templates and understand the available options for SMS pattern detection. 

By separating the retrieval of templates into a method, it provides flexibility and allows for potential customization.